app.rb

```diff
@@ -61,10 +62,17 @@ def create_rental
       puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
     end
     person_number = gets.chomp.to_i
-    print 'Date: '
-    date = gets.chomp
-    @rentals << Rental.new(date, @people[person_number], @books[book_number])
-    puts 'Rental created successfully'
+    print 'Date (YYYY-MM-DD): '
+    date_input = gets.chomp
+    return unless date_input.match(/^\d{4}-\d{2}-\d{2}$/)
+
+    begin
+      date = Date.parse(date_input)
+      @rentals << Rental.new(date, @people[person_number], @books[book_number])
+      puts 'Rental created successfully'
+    rescue ArgumentError
+      puts 'Invalid date format. Please use the format YYYY-MM-DD.'
+    end
```

Comment:
- [ ] Great job so far @tajulafreen But it looks like the error the previous reviewer noticed still persists :wink:  It looks like there is a problem between your `book, person, and rental ` classes as a result the relationships aren't being set properly hence the error :+1: 
- [ ] I noticed that in your `Rental class`, you have something like this below but this `add_rental method` is not accessible in that class hence the error so kindly fix that. You only need to ensure that the rental is adding itself to the `@book` and  `@person` achieving the `belongs-to` association :+1: 
 ```ruby
    @person.add_rental(self)
    @book.add_rental(self)
```
- [ ] Your code can look like this instead :+1: 

```ruby
    @book = book
    # *The rental is adding itself to the @book achieving the 'belongs-to' association
    book.rentals << self

    @person = person
    # *The rental is adding itself to the @person the 'belongs-to' association
    person.rentals << self
  end
```
- [ ] After that, kindly ensure that the methods for `adding rentals` in the `book` and `person` classes are properly setup and working as expected If not, the errors will still persist and currently, I think you need to modify those codes to look something like this below. Please note that this is just an example so kindly tweak it as you deem fit to work properly in your code so that your app can run fully and rentals can be created successfully without any errors :wink: 
```ruby
  def add_rental(person, date)
     Rental.new(date, self, person)
  end
```

- Error Message 
![image](https://github.com/tajulafreen/School_library/assets/102699958/59902ec7-c08b-423a-b13d-7b3e339ce1b9)